types/node/index.d.ts

```diff
@@ -2017,7 +2004,7 @@ declare module "url" {
         hash?: string;
     }
 
-    export function parse(urlStr: string, parseQueryString?: boolean, slashesDenoteHost?: boolean): Url;
+    export function parse(urlStr: string, parseQueryString?: boolean, slashesDenoteHost?: boolean): URL;
```

Comment:
You can only use url.Url with the legacy API and only url.URL with the WHATWG API